src/components/overlays/TimeToExpireBadgeOverlay/TimeToExpireBadgeOverlay.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { useEffect, useRef, useState } from 'react'
+import { Box, styled, Typography } from '@mui/material'
+import { colors } from '../../../index'
+
+const OverlayContainer = styled(Box)<{ width: number; height: number }>(({ width, height }) => ({
+  width: width,
+  height: height,
+  position: 'absolute',
+  top: 0,
+  left: 0,
+  background: 'transparent',
+}))
+
+const TimeLeftContainer = styled(Box)<{ width: string; background: string }>(({ width, background }) => ({
+  display: 'flex',
+  flexDirection: 'column',
+  alignItems: 'center',
+  justifyContent: 'center',
+  background,
+  bottom: 0,
+  position: 'absolute',
+  marginLeft: 'auto',
+  marginRight: 'auto',
+  left: 0,
+  right: 0,
+  width,
+  padding: 8,
+  borderTopLeftRadius: 16,
+  borderTopRightRadius: 16,
+}))
+
+export interface TimeToExpireBadgeOverlayProps {
+  badge: React.ReactNode
+  timeLeft: {
+    quantity: number
+    unitText: string
+  }
+  overlayBackgroundColor?: string // default will be colors.white
+  overlayQuantityColor?: string // default will be colors.black
+  overlayTextColor?: string // default will be colors.greyBackground
+}
+
+export const TimeToExpireBadgeOverlay = (props: TimeToExpireBadgeOverlayProps) => {
+  const [badgeHeight, setBadgeHeight] = useState(0)
+  const [badgeWidth, setBadgeWidth] = useState(0)
+  const [timeToExpireWidth, setTimeToExpireWidth] = useState('0')
+  const badgeRef = useRef(null)
+
+  useEffect(() => {
+    const { clientWidth, clientHeight } = badgeRef?.current || { clientWidth: 0, clientHeight: 0 }
+    setBadgeWidth(clientWidth)
+    setBadgeHeight(clientHeight)
+    setTimeToExpireWidth(clientWidth >= 200 ? '65%' : '85%')
+  })
+
+  return (
+    <Box>
+      <Box ref={badgeRef} sx={{ width: 'fit-content', height: 'fit-content' }}>
+        {props.badge}
+      </Box>
+      <OverlayContainer width={badgeWidth} height={badgeHeight}>
+        {/* time left rectangle */}
+        <Box>
+          <TimeLeftContainer width={timeToExpireWidth} background={props.overlayBackgroundColor || colors.white}>
+            <Typography
+              variant={'h2'}
+              component={'h2'}
+              fontSize={'64px !important'}
```

Comment:
`fontSize` shouldn't be fixed, in the small size badges the text is too big. We must use the default configs for h2 or some other font variant that we already have configured so you don't need to do a custom one for this.
![image](https://github.com/thebadge/thebadge-ui-library/assets/21246763/76084046-8a8f-45f6-a55a-d0afd8f3ca22)

Reply:
There I updated the sizes using the same as for the other overlay. Unfortunately, there is one combination of sizes that it is not in the theme (dAppTitle2 with size 16)